JEasyCryptoServer/src/CryptoServer.java

```diff
@@ -17,16 +17,23 @@
 	private DatagramSocket socket;
 	private byte[] incoming;
 	private boolean running = true;
-	
+	private static int port = 10000;
+
 	public static void main(String[] args) {
+        	if (args.length > 0) {
+        		try {
+				port = Integer.parseInt(args[0]);
+ 			} catch (NumberFormatException e) {
+				e.printStackTrace();
```

Comment:
It would be good, if we had a proper error message. The user might not be able to make sense of only the stack trace. And don't know what is wrong with the input.
How about we provide info about what the program _expected_, _what we got_ and the _overall context_ when it happened?

Reply:
Thank you for your comment, @mbuet2ner!

Would a message like this work?

> 
> PORT is not a correct port. Please provide a correct port.
> 
> Expected, e.g., ./start.sh 3000

Reply:
Then, we could also mention that the default port of 1000 is used instead.

Reply:
Sounds good to me. I added a minor modification. What do you think?

> 

    Error: The PORT you entered is not valid. Please provide a valid port.

    Expected, e.g., './start.sh 3000'

    Falling back to the default port of '10000' instead.

Reply:
Your suggestion sounds much better! I only made a very small adjustment:
> 
> Error: The port ('10.0') you entered is not valid. Please provide a valid port.
> Expected, e.g., './start.sh 3000'
> Falling back to the default port of '10000' instead.

Now the entered port is printed (between the parentheses).